File: java/src/prj/pedrowindisch/empregospcd/handlers/LaunchRequestHandler.java
package prj.pedrowindisch.empregospcd.handlers;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.LaunchRequest;
import com.amazon.ask.model.Response;

import java.util.Optional;

import static com.amazon.ask.request.Predicates.requestType;

public class LaunchRequestHandler implements RequestHandler {
    @Override
    public boolean canHandle(HandlerInput input) {
        return input.matches(requestType(LaunchRequest.class));
    }

    @Override
    public Optional<Response> handle(HandlerInput input) {

        String title = "Empregos PCDs";
        String speechText = "Olá! Para encontrar vagas PCD perto de você, " +
                            "por favor me diga sua cidade.";

        return input.getResponseBuilder()
                .withSpeech(speechText)
                .withSimpleCard(title, speechText)
                .withReprompt(speechText)
                .build();
    }
}

File: nodejs/lambda/handlers/launchRequestHandler.js
const Alexa = require('ask-sdk-core');

module.exports = {
  canHandle(handlerInput) {
    return Alexa.getRequestType(handlerInput.requestEnvelope) === 'LaunchRequest';
  },
  handle(handlerInput) {
    const welcomeMessage =
      "Olá! " +
      "Para encontrar vagas PCD perto de você, por favor me diga sua cidade.";

    const repromptMessage =
      "Por favor, me diga a cidade que você quer procurar vagas."

    return handlerInput.responseBuilder
      .speak(welcomeMessage)
      .reprompt(repromptMessage)
      .getResponse();
  }
};